library/std/src/env.rs

```diff
@@ -298,19 +298,46 @@ impl Error for VarError {
     }
 }
 
+#[unstable(feature = "unsafe_env", issue = "none")]
 /// Sets the environment variable `key` to the value `value` for the currently running
 /// process.
 ///
-/// Note that while concurrent access to environment variables is safe in Rust,
-/// some platforms only expose inherently unsafe non-threadsafe APIs for
-/// inspecting the environment. As a result, extra care needs to be taken when
-/// auditing calls to unsafe external FFI functions to ensure that any external
-/// environment accesses are properly synchronized with accesses in Rust.
+/// # Safety
 ///
-/// Discussion of this unsafety on Unix may be found in:
+/// This function must not be called in the presence of threads. Also, it's not allowed
```

Comment:
```suggestion
/// This function must not be called in the presence of concurrent threads that may simultaneously read or write the environment. Also, it's not allowed
```

(modulo word wrapping)